kubectl-volsync/cmd/pvbackup_create.go

```diff
@@ -0,0 +1,333 @@
+	_, err = pc.ensureReplicationSource(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Wait for ReplicationSource
+	_, err = pc.pr.data.waitForRSStatus(ctx, pc.client)
+	if err != nil {
+		return err
+	}
+
+	// Save the replication source details into relationship file
+	if err = pc.pr.Save(); err != nil {
+		return fmt.Errorf("unable to save relationship configuration: %w", err)
+	}
+
+	// Remove the configuration file saved
+	os.Remove(pc.resticConfig.name)
+	return nil
+}
+
+func (pc *pvBackupCreate) newPVBackupRelationship() *pvBackupRelationshipSource {
+	prs := &pvBackupRelationshipSource{}
+
+	// Assign the values from pvBackupCreate built after parsing cmd args
+	prs.Namespace = pc.Namespace
+	prs.Cluster = pc.Cluster
+	prs.PVCName = pc.SourcePVC
+	prs.RSName = pc.RSName
+	prs.Source.Repository = pc.Repository
+	prs.Trigger.Schedule = &pc.schedule
+
+	return prs
+}
+
+func (pc *pvBackupCreate) ensureSecret(ctx context.Context) error {
+	v := pc.resticConfig.Viper
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pc.Name,
+			Namespace: pc.Namespace,
+		},
+		StringData: map[string]string{
+			"AWS_ACCESS_KEY_ID":     v.Get("AWS_ACCESS_KEY_ID").(string),
+			"AWS_SECRET_ACCESS_KEY": v.Get("AWS_SECRET_ACCESS_KEY").(string),
+			"RESTIC_REPOSITORY":     v.Get("RESTIC_REPOSITORY").(string),
+			"RESTIC_PASSWORD":       v.Get("RESTIC_PASSWORD").(string),
+		},
+	}
+	if err := pc.client.Create(ctx, secret); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (pc *pvBackupCreate) ensureReplicationSource(ctx context.Context) (
+	*volsyncv1alpha1.ReplicationSource, error) {
+	prs := pc.pr.data.Source
+
+	rs := &volsyncv1alpha1.ReplicationSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prs.RSName,
+			Namespace: prs.Namespace,
+		},
+		Spec: volsyncv1alpha1.ReplicationSourceSpec{
+			SourcePVC: prs.PVCName,
+			Trigger: &volsyncv1alpha1.ReplicationSourceTriggerSpec{
+				Schedule: prs.Trigger.Schedule,
+			},
+			Restic: &volsyncv1alpha1.ReplicationSourceResticSpec{
+				Repository: prs.Source.Repository,
+				ReplicationSourceVolumeOptions: volsyncv1alpha1.ReplicationSourceVolumeOptions{
+					CopyMethod: volsyncv1alpha1.CopyMethodClone,
+				},
+			},
+		},
+	}
+
+	if err := pc.client.Create(ctx, rs); err != nil {
+		return nil, err
+	}
+	klog.Infof("Created ReplicationSource: \"%s\" in Namespace: \"%s\" and Cluster: \"%s\"",
+		rs.Name, rs.Namespace, rs.ClusterName)
+
+	return rs, nil
+}
+
+func (prd *pvBackupRelationshipData) waitForRSStatus(ctx context.Context, client client.Client) (
+	*volsyncv1alpha1.ReplicationSource, error) {
+	var (
+		rs  *volsyncv1alpha1.ReplicationSource
+		err error
+	)
+	klog.Infof("waiting for replication source to be available")
+	err = wait.PollImmediate(5*time.Second, defaultRsyncKeyTimeout, func() (bool, error) {
+		rs, err = prd.getReplicationSource(ctx, client)
+		if err != nil {
+			return false, err
+		}
+		// TODO: What should be the condition to break the wait ?
```

Comment:
Need to find a condition that is ideal to say replication Source is ready. Please help me with the proper condition